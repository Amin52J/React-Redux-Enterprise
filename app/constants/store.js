import {routerReducer} from 'react-router-redux';
import {createStore, applyMiddleware} from 'redux';
import {createEpicMiddleware} from 'redux-observable';
import {persistCombineReducers} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import hocReducer from '@hoc/cleanOnUnmount/reducer';
import * as reducers from '@reducers';
import epics from '@epics';
import {ajax} from 'rxjs/observable/dom/ajax';
import {of} from 'rxjs/observable/of';
import 'rxjs/add/operator/map';
import 'rxjs/add/operator/mergeMap';
import 'rxjs/add/operator/catch';
import 'rxjs/add/operator/takeUntil';
import {routerMiddleware} from 'react-router-redux';
import createHistory from 'history/createBrowserHistory';

export const history = createHistory();

const persistConfig = {
  key: 'root',
  storage
};
const epicMiddleware = createEpicMiddleware(epics, {
  dependencies: {
    getJSON: ajax.getJSON,
    of
  }
});
const combinedReducers = persistCombineReducers(persistConfig, {
  ...reducers,
  routing: routerReducer
});
const reducer = hocReducer(combinedReducers);
const store = createStore(reducer, applyMiddleware(epicMiddleware, routerMiddleware(history)));

export default store;
