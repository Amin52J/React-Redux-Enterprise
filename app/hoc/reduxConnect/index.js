// Props Proxy HOC

import {connect} from 'react-redux';

function mapStateToProps (state) {
  return state;
}

export default function (WrappedComponent, actions, states) {
  return connect(states || mapStateToProps, actions)(WrappedComponent);
}
