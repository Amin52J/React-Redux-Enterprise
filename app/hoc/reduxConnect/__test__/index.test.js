import React from 'react';
import {mount} from 'enzyme';
import reduxConnect from '../index';
import renderer from 'react-test-renderer';
import store, {history} from '../../../constants/store';
import {ConnectedRouter} from 'react-router-redux';
import {Provider} from 'react-redux';

class Test extends React.Component {
  render() {
    return (
      <div className="redux-connect">reduxConnect</div>
    );
  }
}

const AttachedTest = reduxConnect(Test);

describe('reduxConnect HOC - HOC', () => {
  it('renders with no problem', () => {
    const component = mount(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <AttachedTest/>
        </ConnectedRouter>
      </Provider>
    );
    const tree = renderer.create(component).toJSON();

    expect(component.find('div').first().hasClass('redux-connect')).toEqual(true);
    expect(tree).toMatchSnapshot();
  });
});
