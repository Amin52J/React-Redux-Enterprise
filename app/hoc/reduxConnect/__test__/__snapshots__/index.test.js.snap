// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reduxConnect HOC - HOC renders with no problem 1`] = `
<div
  className="redux-connect"
>
  reduxConnect
</div>
`;
